apps/common/controllers/base-address.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
+
+const BaseController = require('./base');
+const _ = require('lodash');
+
+module.exports = class BaseAddressController extends BaseController {
+
+  locals(req, res) {
+    const locals = super.locals(req, res);
+    let addresses = this.getAddresses(req);
+    const hasAddresses = !!_.size(addresses);
+    const hasCategories = this.hasCategories(hasAddresses, addresses);
+    if (this.options.addressKey === 'locationAddresses') {
+      addresses = _.filter(addresses, value => value.categories !== undefined);
+    }
+    const items = this.mapAddress(addresses, req);
+
+    return Object.assign({}, locals, {
+      items,
+      hasAddresses,
+      hasCategories
+    });
+  }
+
+  getAddresses(req) {
+    return req.sessionModel.get(this.options.addressKey);
+  }
```

Comment:
I think @lennym was correct, this method is unnecessary. The return value from `req.sessionModel.get` can be implicitly tested from `locals()`